src/map/utils/battleutils.cpp

```diff
@@ -2630,15 +2630,23 @@ namespace battleutils
 
                 if (isAvatar)
                 {
+                    dLvl = PAttacker->GetMLevel() - PDefender->GetMLevel();
+
                     if (dLvl > 0)
                     {
                         // Avatars have a known level difference cap of 38
                         hitrate += static_cast<int16>(std::min(dLvl, (int16)38) * 2);
                     }
                 }
-                else
+
+                // Only players are penalized for dLvl
+                if (PAttacker->objtype == TYPE_PC)
                 {
-                    // Everything else has no known caps, though it's likely 38 like avatars
+                    if (PAttacker->GetMLevel() < PDefender->GetMLevel())
+                    {
+                        dLvl = PAttacker->GetMLevel() - PDefender->GetMLevel();
+                    }
+
```

Comment:
If you leave the dLvl definition like it was (directly set the level diff) you dont need to add it to avatars.

Then, the new block of code would only need to be...
```c++
// Only players are penalized for dLvl
if (PAttacker->objtype == TYPE_PC)
{
    if (dLvl > 0)
    {
        dLvl = 0;
    }
}
```

Reply:
Unless, of course, you have future edits in mind for this.

Reply:
You are right - I originally had something like this and then changed it back to its own if statement. I could do something like this as well - whichever you prefer. 
```cpp
if (shouldApplyLevelCorrection)
{
    int16 dLvl = 0;
    bool isPet = PAttacker->objtype == TYPE_PET;
    bool isAvatar = false;

    if (isPet)
    {
        CPetEntity* petEntity = dynamic_cast<CPetEntity*>(PAttacker);
        isAvatar = petEntity && petEntity->getPetType() == PET_TYPE::AVATAR;
    }

    if (isAvatar)
    {
        dLvl = PAttacker->GetMLevel() - PDefender->GetMLevel();
        dLvl = std::min(dLvl, static_cast<int16>(38));
    }
    else if (PAttacker->objtype == TYPE_PC && PAttacker->GetMLevel() < PDefender->GetMLevel())
    {
        dLvl = PAttacker->GetMLevel() - PDefender->GetMLevel();
    }

    hitrate += static_cast<int16>(dLvl * 2);
}
```

Reply:
```c++
if (shouldApplyLevelCorrection)
{
    int16 dLvl     = PAttacker->GetMLevel() - PDefender->GetMLevel();
    bool isAvatar = false;

    if (PAttacker->objtype == TYPE_PET)
    {
        CPetEntity* petEntity = dynamic_cast<CPetEntity*>(PAttacker);
        isAvatar = petEntity && petEntity->getPetType() == PET_TYPE::AVATAR;
    }

    if (isAvatar)
    {
        dLvl = std::min(dLvl, static_cast<int16>(38));
    }
    else if (PAttacker->objtype == TYPE_PC && dLvl > 0)
    {
        dLvl = 0;
    }

    hitrate += static_cast<int16>(dLvl * 2);
}
```

My reasoning behind this, other than being marginaly shorter, is any case where entities arent players or avatars, i dont think dlvl should be 0. I may be wrong though.

Reply:
If we leave that defined on top then if the entity is a Mob lets say - regardless of its higher or lower when attacking a player there would be some sort of level correction by dLvl.

From the research jimmayus did on mob pdif/acc:
Monster hit rate is: 75 - Floor((acc-eva)/2), no level correction
-->> *there might be positive hit correction*

So in either case the dLvl should be clamping at 0 for mobs regardless. If we can find more proof that there is positive hit correction (like there is for PDIF) then we can add that in later. As for now we have no solid evidence of that until more testing has been done.